### ISSUE TITLE
Fixed attributes in XSD are not enforced

### ISSUE THREAD

**Issue by ruffsl** [brunato/xmlschema#26]
Fixating an attribute's value in the xsd schema has no effect when validating xml data.

A minimal example:
https://gist.github.com/ruffsl/db63d2d0c9845ae3b1c4a444c01bfd12

xsd_string =
``` xml
<?xml version="1.0" encoding="UTF-8"?>
<xs:schema xmlns:xs="http://www.w3.org/2001/XMLSchema"
    elementFormDefault="qualified"
    attributeFormDefault="unqualified">

    <xs:element name="foo" type="Foo" />

    <xs:complexType name="Foo">
        <xs:sequence minOccurs="1" maxOccurs="unbounded">
            <xs:element name="bar" type="Bar" />
        </xs:sequence>
    </xs:complexType>

    <xs:complexType name="Bar">
        <xs:sequence minOccurs="1" maxOccurs="1">
            <xs:element name="subject_name" type="xs:string" />
        </xs:sequence>
        <xs:attribute name="name" fixed="BAR" use="required"/>
    </xs:complexType>
</xs:schema>
``` 

xml_string_1 = 
```xml
<?xml version="1.0" encoding="UTF-8"?>
<foo>
    <bar name="BAR">
        <subject_name>Bar</subject_name>
    </bar>
</foo>
```

xml_string_2 = 
``` xml
<?xml version="1.0" encoding="UTF-8"?>
<foo>
    <bar name="RAB">
        <subject_name>Bar</subject_name>
    </bar>
</foo>
```

xml_string_3 = 
``` xml
<?xml version="1.0" encoding="UTF-8"?>
<foo>
    <bar>
        <subject_name>Bar</subject_name>
    </bar>
</foo>
```

``` python
>>> import xmlschema
>>> xsd_schema = xmlschema.XMLSchema(xsd_string)
>>> print(xsd_schema.validate(data=xml_string_1, use_defaults=False))
None
>>> print(xsd_schema.validate(data=xml_string_2, use_defaults=False))
None
>>> print(xsd_schema.validate(data=xml_string_3, use_defaults=False))
---------------------------------------------------------------------------
XMLSchemaValidationError                  Traceback (most recent call last)
<ipython-input-8-cc56bcfbbb75> in <module>()
----> 1 print(xsd_schema.validate(data=xml_string_3, use_defaults=False))

/usr/local/lib/python3.5/dist-packages/xmlschema/xsdbase.py in validate(self, data, use_defaults)
    248         """
    249         for error in self.iter_errors(data, use_defaults=use_defaults):
--> 250             raise error
    251 
    252     def iter_errors(self, data, path=None, use_defaults=True):

XMLSchemaValidationError: failed validating {} with <XsdAttributeGroup at 0x7f8c506dad30>.

Reason: missing required attributes {'name'}

Schema:

  <xs:complexType xmlns:xs="http://www.w3.org/2001/XMLSchema" name="Bar">
        <xs:sequence maxOccurs="1" minOccurs="1">
            <xs:element name="subject_name" type="xs:string" />
        </xs:sequence>
        <xs:attribute fixed="BAR" name="name" use="required" />
  </xs:complexType>

Instance:

  <bar>
        <subject_name>Bar</subject_name>
  </bar>
```

Note that a `XMLSchemaValidationError` properly occurs when the required attribute is omitted entirely, but not when the attribute value differs from what is expected/fixated in the schema.

**Comment by brunato** [comment]
This can be fixed with a patch for `XsdAttribute.iter_decode` method: 

```python
def iter_decode(self, text, validation='lax', **kwargs):
    if not text and kwargs.get('use_defaults', True):
        text = self.default
    if self.fixed and text != self.fixed:  # Added a fixed validation!!
        yield XMLSchemaValidationError(self, text, "value differs from fixed value")

    for result in self.type.iter_decode(text, validation, **kwargs):
        yield result
        if not isinstance(result, XMLSchemaValidationError):
            return
```

**Comment by brunato** [comment]
This issue should be fixed in the new version 0.9.15.
